### DISCUSSION TITLE
Teamspeak Server

### DISCUSSION THREAD

**Discussion by Stormrage-DJ** [alexbelgium/hassio-addons#342] [Q&A] [answered]
Hi There!

Is it possible somehow to access the teamspeak files via samba share? What I would like to do is to get the privilege key somehow and also to restore my old database from my TrueNAS into this instance!

Any idea is welcome! Thanks!

**Comment by alexbelgium** [comment]
Hi, thanks I'll look at that

**Comment by alexbelgium** [comment]
The data is stored in the folder /data. Possible solutions : use portainer to access the folder (easiest), use ssh to access the addon folder (needs config), or i adapt the addon (longest).

Perhaps using portainer would be the easiest for you... What do you think?

**Reply by Stormrage-DJ** [reply]
I wouldn't mind some further support. I've installed Portrainer and tried to explore the menus and try to find how to access my ts3server instance, however I only could find the docker container and image, but couldn't access the contents.

![image](https://user-images.githubusercontent.com/83466158/170825721-78528d15-b4f5-4016-a0df-e81a0b27822c.png)

I also have SSH access to my Home Assistant server, however I failed to find the ts3 instance. One thing i could certainly identify is the "config" folder that you can also access via samba share.

![image](https://user-images.githubusercontent.com/83466158/170825670-f187d322-d5c7-4ff1-bb8f-489a4e6e4be6.png)

Any idea is very much appreciated!

**Comment by alexbelgium** [comment]
Hi, I don't have much time for an elaborate answer, sorry about that I'll look at it a bit later. In portainer, you need to click on >_ console, which will open a bash console with user root. It's just below the orange "business edition feature". Once there, all remanent data from the addon is stored in /data. For other addons I sometime indeed use shared folders such as /config or /share but I hadn't enabled this for this addon. 

With ssh I don't remember the exact location but it's a bit more complicated : we shouldn't ssh to the homeassistant container (as only data from HA itself would be available), but to the underlying os using a protocol that includes copying some files with an USB stick, and then find the folder where the addon stores its data... Portainer achieves the same but much more easily :) 

This evening or tomorrow I'll have a closer look!

**Reply by Stormrage-DJ** [reply]
With console I could open it, tho no copy-paste is possible from external sources, i assume it is not ment by design of docker concept, but rather these config files, license file (if avilable) and database somehow shall be linked to an external location of the host like the regular /config.

Anyways, not in a hurry, but on the long run I would move my ts3server from truenas into home assistant and release its virtual ubuntu.

**Comment by alexbelgium** [comment, marked as answer]
Hi, actually the /share folder is accessible from within the addon, and this folder is a shared one through all homeassistant. For example, if you use the samba share addon (from the community store), it exposes the /share folder through SMB. With the console, you can copy the files you need from /share to /data. Would that allow what you are aiming to do?

**Reply by Stormrage-DJ** [reply]
Thanks! I could do what I wanted, restored/migrated TS3 config/database from my other server! :)

**Reply by alexbelgium** [reply]
Glad it worked!
